components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -364,10 +364,6 @@ class GeckoEngineSession(
             session: GeckoSession,
             request: NavigationDelegate.LoadRequest
         ): GeckoResult<AllowOrDeny> {
-            if (request.target == NavigationDelegate.TARGET_WINDOW_NEW) {
-                return GeckoResult.fromValue(AllowOrDeny.ALLOW)
-            }
```

Comment:
One difference here is that, before when handling a new window request, we didn't notify observers of `onLoadRequest`, now we will notify. Did you check the consequences of that? There's two observers in Fenix which seem fine? One is for telemetry and one for "expanding" the toolbar.

Reply:
Agreed, I'll double check Fenix side usage.

Reply:
Looking more into this.  You're right, I don't think we should notify the observers when `TARGET_WINDOW_NEW` is set as the target.  I'll add an extra check before notifying the observer.  Thanks,